ledger/db/src/lib.rs

```diff
@@ -1022,182 +1104,1166 @@ mod ledger_db_test {
+                    mint_config: set_mint_config_tx1.prefix.configs[0].clone(),
+                    total_minted: mint_tx1.prefix.amount,
+                },
+                ActiveMintConfig {
+                    mint_config: set_mint_config_tx1.prefix.configs[1].clone(),
+                    total_minted: 0,
+                }
+            ]
+        );
+
+        // === Sanity: Allow the second mint configuration to match, which
+        // should allow minting to succeeed.
+        let mint_tx3 = generate_test_mint_tx(token_id1, &signers1, 11, &mut rng);
+
+        let block_contents3 = BlockContents {
+            mint_txs: vec![mint_tx3.clone()],
+            ..Default::default()
+        };
+
+        let block3 = Block::new_with_parent(
+            BLOCK_VERSION,
+            &block2,
+            &Default::default(),
+            &block_contents3,
+        );
+
+        ledger_db
+            .append_block(&block3, &block_contents3, None)
+            .unwrap();
+
+        // Amount minted should not update.
+        assert_eq!(
+            ledger_db.get_active_mint_configs(token_id1).unwrap(),
+            vec![
+                ActiveMintConfig {
+                    mint_config: set_mint_config_tx1.prefix.configs[0].clone(),
+                    total_minted: mint_tx1.prefix.amount,
+                },
+                ActiveMintConfig {
+                    mint_config: set_mint_config_tx1.prefix.configs[1].clone(),
+                    total_minted: 11,
+                }
+            ]
+        );
+    }
+
+    #[test]
+    #[should_panic(expected = "called `Result::unwrap()` on an `Err` value: NoOutputs")]
+    // Appending an empty block should fail.
+    fn test_append_block_fails_when_block_is_empty() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let mut ledger_db = create_db();
+
+        // === Create and append the origin block. ===
+        // The origin block contains a single output belonging to the
+        // `origin_account_key`.
+
+        let origin_account_key = AccountKey::random(&mut rng);
+        let (origin_block, origin_block_contents) =
+            get_origin_block_and_contents(&origin_account_key);
+
+        ledger_db
+            .append_block(&origin_block, &origin_block_contents, None)
+            .unwrap();
+
+        // === Append a block with no contents. ===
+
+        let block_contents1 = Default::default();
+
+        let block1 = Block::new_with_parent(
+            BLOCK_VERSION,
+            &origin_block,
+            &Default::default(),
+            &block_contents1,
+        );
+
+        ledger_db
+            .append_block(&block1, &block_contents1, None)
+            .unwrap();
+    }
+
+    #[test]
+    #[should_panic(expected = "called `Result::unwrap()` on an `Err` value: NoKeyImages")]
+    // Appending a non-origin block should fail if the block contains no key images.
+    fn test_append_block_fails_for_non_origin_blocks_without_key_images() {
```

Comment:
Does this apply to blocks that only contain minting or mint configuration?

Reply:
No, if they contain minting/mint configuration txs then it wont fail. I can change the test name to reflect that